src/evict/evict_lru.c

```diff
@@ -702,18 +704,34 @@ __evict_update_work(WT_SESSION_IMPL *session)
         total_dirty = total_inmem = total_updates = 0;
         hs_id = 0;
         for (;;) {
-            WT_RET_NOTFOUND_OK(ret = __wt_curhs_next_hs_id(session, hs_id, &hs_id));
+            ret = __wt_curhs_next_hs_id(session, hs_id, &hs_id);
+            WT_ASSERT(session, ret == 0 || ret == WT_NOTFOUND);
```

Comment:
You may want to assert `hs_id` is less than the max value.

Reply:
Yes, that should be enough but I am thinking of a new way to walk through multiple HS. I might do it in the next ticket. Still investigating.